app/src/org/commcare/logging/DataChangeLogger.kt

```diff
@@ -0,0 +1,155 @@
+package org.commcare.logging
+
+import android.content.Context
+import android.os.Environment
+import android.util.Log
+import org.apache.commons.lang3.StringUtils
+import org.commcare.android.logging.ReportingUtils
+import org.commcare.utils.CrashUtil
+import org.commcare.utils.FileUtil
+import org.javarosa.core.io.StreamsUtil
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.IOException
+import java.text.SimpleDateFormat
+import java.util.*
+
+/**
+ *  Used to log some crtitical platform events like DB migrations, Clear user data,
+ *  app installations etc. All these logs are backed by file storage and are accessible
+ *  from outside CommCare
+ */
+class DataChangeLogger {
+
+    fun init(context: Context) {
+        initLogFiles(context)
+    }
+
+    private fun initLogFiles(context: Context) {
+        if (!isExternalStorageWritable()) {
+            Log.e(TAG, "External Storage unavialable to write logs")
+            return
+        }
+
+        primaryFile = initFile(context, PRIMARY_LOG_FILE_NAME)
+        if (primaryFile!!.exists() && primaryFile!!.length() > MAX_FILE_SIZE) {
+            // When primary file gets full -> Delete secondary and move logs in primary to secondary
+            secondaryFile = initFile(context, SECONDARY_LOG_FILE_NAME)
+            secondaryFile!!.delete()
+            try {
+                FileUtil.copyFile(primaryFile, secondaryFile)
+                primaryFile!!.delete()
+                primaryFile!!.createNewFile()
+            } catch (e: IOException) {
+                e.printStackTrace()
+            }
+        }
+    }
+
+    private fun initFile(context: Context, fileName: String): File {
+        val file = File(context.getExternalFilesDir(null), fileName)
+        if (!file.exists()) {
+            try {
+                file.createNewFile()
+            } catch (e: IOException) {
+                e.printStackTrace()
+            }
+        }
+        return file
+    }
+
+    companion object {
+
+        private const val PRIMARY_LOG_FILE_NAME = "CommCare Data Change Logs.txt"
+        private const val SECONDARY_LOG_FILE_NAME = "CommCare Data Change Logs1.txt"
+        private const val DATE_FORMAT = "d MMM yy HH:mm:ss z"
+        private const val MAX_FILE_SIZE = (40 * 1000).toLong() // ~40KB
```

Comment:
Curious how you picked this?

Reply:
I did a basic byte calculation for 200 logs assuming each to be 100 characters long.

Reply:
Makes sense, though we may want to bump up the number of logs we plan to keep around. If we expect each file to fit ~200 logs, that gives us a history of 400 total. I think the number Clayton had proposed was closer to 1,000 logs, so maybe we can do 500 per file?